### ISSUE TITLE
Faker class generate method with default locale

### ISSUE THREAD

**Issue by Ramon5** [FactoryBoy/factory_boy#790]
#### The problem
I have a old version in my project, then using Faker class, i generate faker name with Faker("name") but in newer versions i have using Faker("name").generate(), thats problem is, when i using the generate method without params, the 'locale' key error is getted

#### Proposed solution

If the developer does not provide a dictionary with a locale, a default locale is provided.

```
import locale as local_language

class Faker(declarations.ParameteredDeclaration):

        ...

        def generate(self, params: dict = None):
             locale = params.pop('locale', None) or {'locale': local_language.getlocale()[0]}
             subfaker = self._get_faker(locale)
             return subfaker.format(self.provider, **params)
```

#### Extra notes
factory_boy version 3.1.0

**Comment by rbarrois** [comment]
Hi! Thanks for the report.

There is indeed something wrong with the latest release, but I haven't found the time to look into it — see #788 #787 #786 #785…

However, please note that the `factory.Faker(...).generate()` method is private.

As described [here](https://github.com/FactoryBoy/factory_boy/pull/785#issuecomment-703089052), the `params` dict should have been properly prepared before reaching the `.generate()` function; but that fails in specific cases — with `factory.Maybe` and `class Params`.

**Comment by tomasgarzon** [comment]
Any update here? when do you plan to move this to pypi package? (kwargs.pop('locale') keyerror)

**Comment by rbarrois** [comment]
This has been fixed in #828; a new release should be made in the next couple of days :)
